pandas/core/dtypes/cast.py

```diff
@@ -1078,8 +1080,9 @@ def convert_dtypes(
                 inferred_dtype = input_array.dtype
 
         if convert_boolean:
-            if is_bool_dtype(input_array.dtype) and not is_extension_array_dtype(
-                input_array.dtype
+            if is_bool_dtype(input_array.dtype) and (
+                (not is_extension_array_dtype(input_array.dtype))
+                or (input_array.dtype == pd.BooleanDtype())
```

Comment:
Can this just be simplified to `is_bool_dtype`?

Reply:
I think the issue is actually earlier. The condition at https://github.com/pandas-dev/pandas/pull/32490/files#diff-1df6c746d2a342e60cce09f774b566d7R1073-R1076 isn't wanted here.

Should this entire check be inside a big `if not is_extension_array_dtype(input_array.dtype)`? There aren't any cases where we want to convert from extension type to extension type, right?

Reply:
And by "entire check" I mean immediately following the `if convert_string or convert_integer or convert_boolean` at https://github.com/pandas-dev/pandas/pull/32490/files#diff-1df6c746d2a342e60cce09f774b566d7R1054